BinaryTreeRange.java

```diff
@@ -0,0 +1,70 @@
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class BinaryTreeRange {
+    public static void main(String[] args) {
+        test(new int[]{1,2,3,4,5}, 1, 5);
+        test(new int[]{}, 1, 5);
+        test(new int[]{1,2,3,4,5}, 5, 3);
+    }
+
+    public static int findNumbersInRange(Node root, int start, int end) {
+        int result = 0;
+        Queue<Node> notVisited = new LinkedList<>();
+        if(root == null) return 0;
+        notVisited.add(root);
+
+        while(!notVisited.isEmpty()) {
+            Node current = notVisited.remove();
+            if(current.data >= start && current.data <= end) {
+                result++;
+            }
+
+            if(current.left != null) notVisited.add(current.left);
+            if(current.right != null) notVisited.add(current.right);
+        }
```

Comment:
The algorithm for Breadth First Search looks good.

Such a problem will evoke a good discussion about DFS vs BFS:
1. DFS uses O(D) memory where D is the depth of the tree.
2. BFS uses O(W) memory where W is the maximum width of the tree.

If the input trees are reasonably balanced then their depth will be closer to O(log N) and their width will be O(N). Of course, the more the trees are imbalance the higher the depth will be. But typically binary trees are used in places where it is possible to balance to take advantage of O(log N) complexity. So it is better to use DFS in most problems which just require traversing all the nodes of a tree but the order in which they are traversed in not important.